lago/export.py

```diff
@@ -0,0 +1,239 @@
+            self.exported_metadata['size'] = os.stat(self.dst).st_size
+            self.exported_metadata['name'] = self.name
+            self.exported_metadata['version'] = time.strftime("%Y%m%d.0")
+
+    def write_lago_metadata(self):
+        with LogTask('Writing Lago metadata'):
+            with open(self.dst + '.metadata', 'wt') as f:
+                json.dump(self.exported_metadata, f)
+
+
+class TemplateExportManager(DiskExportManager):
+    """
+    TemplateExportManager is responsible exporting images of type template.
+
+    Attributes:
+        standalone (bool): If true, create a new image which is the result of
+         merging all the layers of src (the image that we want to export).
+        src_qemu_info (dict): Metadata on src which was generated by qemu-img.
+    """
+
+    def __init__(self, dst, disk_type, disk, do_compress, *args, **kwargs):
+        super(TemplateExportManager, self).__init__(
+            dst, disk_type, disk, do_compress
+        )
+        self.standalone = kwargs['standalone']
+        self.src_qemu_info = utils.get_qemu_info(self.src, backing_chain=True)
+
+    def rebase(self):
+        """
+        Change the backing-file entry of the exported disk.
+        Please refer to 'qemu-img rebase' manual for more info.
+        """
+        if len(self.src_qemu_info) == 1:
+            # Base image (doesn't have predecessors)
+            return
+
+        if self.standalone:
+            # Consolidate the layers and base image
+            utils.qemu_rebase(target=self.dst, backing_file="")
+        else:
+            if len(self.src_qemu_info) > 2:
+                LOGGER.error(
+                    'Layered export is currently supported for one '
+                    'layer only.  You can try to use Standalone export.'
+                )
+                sys.exit(1)
+            # Put an identifier in the metadata of the copied layer,
+            # this identifier will be used later by Lago in order
+            # to resolve and download the base image
+            parent = self.src_qemu_info[0]['backing-filename']
+            parent = './{}'.format(os.path.basename(parent.split(':', 1)[1]))
+            utils.qemu_rebase(target=self.dst, backing_file=parent, safe=False)
+
+    def update_lago_metadata(self):
+        super(TemplateExportManager, self).update_lago_metadata()
+
+        if self.standalone:
+            self.exported_metadata['base'] = 'None'
+        else:
+            self.exported_metadata['base'] = os.path.basename(self.src)
+
+    def export(self):
+        with LogTask('Exporting disk {} to {}'.format(self.name, self.dst)):
+            with utils.RollbackContext() as rollback:
+                rollback.prependDefer(
+                    shutil.rmtree, self.dst, ignore_errors=True
+                )
+                self.copy()
+                self.sparse()
+                self.rebase()
+                self.calc_sha('sha1')
+                self.calc_sha('sha512')
```

Comment:
I followed the convention that is being used in Lago Images, even though we don't really use 'sha512'. The verification of the image is done with  'sha128'. If we decide to drop 'sha512' in the export feature we also need to drop it from Lago Images (I don't mind to drop it). 
What do you think ?